package.json

```diff
@@ -185,6 +182,28 @@
     "winston": "^3.3.3",
     "ws": "^7.5.6"
   },
+  "optionalDevDependencies": {
+    "kafka-avro": {
+      "engine": "< 16",
+      "version": "^3.1.1"
+    },
+    "pg-native": {
+      "engine": "< 18",
+      "version": "^3.0.0"
+    },
+    "node-rdkafka": {
+      "engine": "< 18",
+      "version": "^2.12.0"
+    },
+    "grpc": {
+      "engine": "< 18",
+      "version": "^1.24.11"
+    },
+    "mali": {
+      "engine": "< 18",
+      "version": "0.23.0"
+    }
+  },
```

Comment:
_praise:_ this new `optionalDevDependencies` section and the mechanism behind it is awesome 🤩